lib/ansible/plugins/inventory/aws_ec2.py

```diff
@@ -72,6 +74,11 @@
           type: bool
           default: False
           version_added: '2.8'
+        allow_unsafe_hostnames:
```

Comment:
doesn't this overlap with `use_contrib_script_compatible_sanitization` ?

Reply:
Yes, it does overlap with the regex to use. The hostname sanitization is rather sparse in the contrib script, so sanitizing everything depending on the setting of `use_contrib_script_compatible_sanitization` doesn't replicate the script behavior. I could instead add a whitelist of the hostnames that were sanitized and use that in conjunction with use_contrib_script_compatible_sanitization. Is deprecating the current behavior the right way to go? My last commit reverted that because I'm not sure how important it is, but I can drop that commit.

Reply:
@bcoca Or rather than just fixing aws_ec2, maybe I could add it to lib/ansible/inventory/host.py similarly to what you did for groups and share common code in inventory/helpers?